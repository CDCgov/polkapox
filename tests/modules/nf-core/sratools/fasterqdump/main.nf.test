nextflow_process {

    name "Test Process SRATOOLS_FASTERQDUMP"
    script "modules/nf-core/sratools/fasterqdump/main.nf"
    process "SRATOOLS_FASTERQDUMP"

    test("Fasterq Dump Test") {

        when {
            params {
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'SRR29268321'],
                    [
                        file("${projectDir}/assets/test_sra_list", checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
