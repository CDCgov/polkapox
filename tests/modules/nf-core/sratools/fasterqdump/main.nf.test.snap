{
    "Fasterq Dump Test": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "forward": [
                    
                ],
                "reverse": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-11T16:22:05.212605033"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "forward": [
                    
                ],
                "reverse": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-09T13:20:10.632106568"
    }
}